index.d.ts

```diff
@@ -166,7 +166,7 @@ export class Client {
 export class Downloader {
 	constructor();
 
-	downloadHLS(url: string, options?: m3u8Options): Promise<m3u8Stream>;
+	downloadHLS(url: string, options?: m3u8Options): m3u8Stream;
```

Comment:
This was `Promise<m3u8Stream>` to align with `progressive` stream which is `Promise<IncomingMessage>`

Reply:
But it doesn't actually need a promise in Downloader. What should I do? Just change typing?
However `Song.downloadHLS` is still return Promise<m3u8Stream>.

Reply:
we could also update `downloadProgressive` to not return a promise by doing something like:

```js
const outStream = new PassThrough();

setImmediate(() =>
    req.get(url, res => {
        res.pipe(outStream);
    });
});

return outStream;
```

Reply:
I see.
How about other code in my PR? Is it OK?

Reply:
Also we should handle 'error' event of req.

Reply:
> Also we should handle 'error' event of req.get's return value.

Instead, we should forward `error` to `outStream` and let users handle it

Reply:
Then, Will you change it from this PR to that way?